core/src/main/java/de/danoeh/antennapod/core/service/download/DownloadService.java

```diff
@@ -639,4 +670,19 @@ private void postDownloaders() {
                     new PostDownloaderTask(downloads), 1, 1, TimeUnit.SECONDS);
         }
     }
+
+    public static boolean appIsRunning(@NonNull Context context) {
```

Comment:
So... I found this snippet on StackOverflow. I wonder if it's maybe a bit... not nice to call every second while downloading from a background DownloadService, even if it does appear to work.

It came from here, where there are a variety of options presented. I was more focused on getting something that worked rather than on the best possible solution, so maybe one of the alternatives would actually work out better (eg. the LifecycleObserver seems like the "best" way to do it now).
https://stackoverflow.com/questions/8489993/check-android-application-is-in-foreground-or-not/